pkg/backup/compaction_job.go

```diff
@@ -401,6 +421,35 @@ func (b *backupResumer) ResumeCompaction(
 	return b.processScheduledBackupCompletion(ctx, jobs.StateSucceeded, execCtx, updatedDetails)
 }
 
+func (b *backupResumer) processCompactionCompletion(
+	ctx context.Context,
+	execCtx sql.JobExecContext,
+	env scheduledjobs.JobSchedulerEnv,
+	details jobspb.BackupDetails,
+) error {
+	if details.ScheduleID != 0 {
```

Comment:
uber nit: this function would be slightly cleaner if you rearranged:
```
if details.ScheduleID == 0 {
  return nil
}
return execCfg().Txn()...
```

Reply:
Oh yup, if guards are much better